### PR TITLE
use token roles

### PR DESCRIPTION
**Pull request by alavieille**
https://github.com/open-orchestra/open-orchestra-media-admin-bundle/pull/329
https://github.com/open-orchestra/open-orchestra-cms-bundle/pull/2031

### CODE REVIEW

File: Backoffice/Context/ContextManager.php [hunk 1]
```diff
@@ -26,26 +28,30 @@ class ContextManager implements CurrentSiteIdInterface
...
+     * @param SiteRepositoryInterface       $siteRepository
+     * @param AuthorizationCheckerInterface $authorizationChecker
      */
     public function __construct(
         Session $session,
         TokenStorageInterface $tokenStorage,
         $defaultLocale,
-        SiteRepositoryInterface $siteRepository
+        SiteRepositoryInterface $siteRepository,
+        AuthorizationCheckerInterface $authorizationChecker
```
Comment (ngilain) [inline Backoffice/Context/ContextManager.php]: @alavieille : By the way, can you update the old code `) {` on the same line ?

File: Backoffice/Security/Authorization/Voter/AbstractNodeVoter.php [hunk 2]
```diff
@@ -25,45 +25,50 @@
...
+     * @param NodeInterface  $node
+     * @param TokenInterface $token
      *
      * @return bool
      */
-    protected function voteForReadAction($node, UserInterface $user)
+    protected function voteForReadAction($node, TokenInterface $token)
     {
-        return $user->hasRole(ContributionRoleInterface::NODE_CONTRIBUTOR)
+        $user = $token->getUser();
```
Comment (ngilain) [inline Backoffice/Security/Authorization/Voter/AbstractNodeVoter.php]: @alavieille : there's no more the need to create the local var $user, you can directly inject $token->getUser() in the condition

File: Backoffice/Security/Authorization/Voter/AbstractNodeVoter.php [hunk 3]
```diff
@@ -25,45 +25,50 @@
...
      *
      * @return bool
      */
-    protected function voteForOwnedSubject($action, $node, UserInterface $user)
+    protected function voteForOwnedSubject($action, $node, TokenInterface $token)
     {
-        return $user->hasRole(ContributionRoleInterface::NODE_CONTRIBUTOR)
+        $user = $token->getUser();
+
+        return $this->hasRole($token, ContributionRoleInterface::NODE_CONTRIBUTOR)
```
Comment (ngilain) [inline Backoffice/Security/Authorization/Voter/AbstractNodeVoter.php]: @alavieille : there's no more the need to create the local var $user, you can directly inject $token->getUser() in the condition

File: Backoffice/Security/Authorization/Voter/AbstractNodeVoter.php [hunk 4]
```diff
@@ -25,45 +25,50 @@
...
+     * @param NodeInterface  $node
+     * @param TokenInterface $token
      *
      * @return bool
      */
-    protected function voteForOwnedSubject($action, $node, UserInterface $user)
+    protected function voteForOwnedSubject($action, $node, TokenInterface $token)
     {
-        return $user->hasRole(ContributionRoleInterface::NODE_CONTRIBUTOR)
+        $user = $token->getUser();
```
Comment (ngilain) [inline Backoffice/Security/Authorization/Voter/AbstractNodeVoter.php]: @alavieille : there's no more the need to create the local var $user, you can directly inject $token->getUser() in the condition, or better, you can inject the $token in `isSubjectInPerimeter`. That way if we refacto the token to store the perimeter, the modification here will be small

File: Backoffice/Security/Authorization/Voter/AbstractNodeVoter.php [hunk 5]
```diff
@@ -25,45 +25,50 @@
...
+     * @param string         $action
+     * @param NodeInterface  $node
+     * @param TokenInterface $token
      *
      * @return bool
      */
-    protected function voteForSomeoneElseSubject($action, $node, UserInterface $user)
+    protected function voteForSomeoneElseSubject($action, $node, TokenInterface $token)
     {
+        $user = $token->getUser();
```
Comment (ngilain) [inline Backoffice/Security/Authorization/Voter/AbstractNodeVoter.php]: @alavieille : there's no more the need to create the local var $user, you can directly inject $token->getUser() in the condition, or better, you can inject the $token in `isSubjectInPerimeter`. That way if we refacto the token to store the perimeter, the modification here will be small

File: Backoffice/Security/Authorization/Voter/ContentVoter.php [hunk 6]
```diff
@@ -36,12 +36,13 @@ protected function supportSubject($subject)
...
      * @param ContentInterface|string $subject
-     * @param UserInterface           $user
+     * @param TokenInterface          $token
      *
      * @return bool
      */
-    protected function voteForReadAction($subject, UserInterface $user)
+    protected function voteForReadAction($subject, TokenInterface $token)
     {
+        $user = $token->getUser();
```
Comment (ngilain) [inline Backoffice/Security/Authorization/Voter/ContentVoter.php]: @alavieille : there's no more the need to create the local var $user, you can directly inject $token->getUser() in the condition, or better, you can inject the $token in `isSubjectInPerimeter`. That way if we refacto the token to store the perimeter, the modification here will be small

File: Backoffice/Security/Authorization/Voter/ContentVoter.php [hunk 7]
```diff
@@ -72,16 +75,17 @@ protected function voteForOwnedSubject($action, $content, UserInterface $user)
...
      * @return bool
      */
-    protected function voteForSomeoneElseSubject($action, $subject, UserInterface $user)
+    protected function voteForSomeoneElseSubject($action, $subject, TokenInterface $token)
     {
         if (!is_object($subject)) {
             return false;
         }
 
+        $user = $token->getUser();
```
Comment (ngilain) [inline Backoffice/Security/Authorization/Voter/ContentVoter.php]: @alavieille : there's no more the need to create the local var $user, you can directly inject $token->getUser() in the condition, or better, you can inject the $token in `isSubjectInPerimeter`. That way if we refacto the token to store the perimeter, the modification here will be small

**Comment by alavieille** [comment]
@ngilain  pr updated

**Comment by ngilain** [comment]
:+1:

**Pull request merged**